src/evergreen/api.py

```diff
@@ -159,7 +159,9 @@ def _raise_for_status(response: requests.Response) -> None:
 
         response.raise_for_status()
 
-    def _paginate(self, url: str, params: Dict = None) -> Dict:
+    def _paginate(
+        self, url: str, params: Dict = None
+    ) -> Union[List[Dict[str, Any]], Dict[str, Any]]:
```

Comment:
There are many mypy overrides in this file based on what this is returning. It may make sense to add a refactoring ticket to cast them properly rather than just telling mypy to hush.